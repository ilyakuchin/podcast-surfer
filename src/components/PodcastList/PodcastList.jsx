import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Loader, Dimmer, Card, Image } from 'semantic-ui-react';
import { fetchCurrentPodcast } from '../../redux/actions/CurrentPodcast/currentPodcast';

export function PodcastList({
  isFetching,
  podcasts,
  fetchCurrentPodcastConnect
}) {
  return !isFetching ? (
    <Card.Group>
      {podcasts.map(({ name, imageUrl, rss }) => (
        <Card centered key={rss}>
          <Card.Content textAlign='center'>
            <Image
              style={{
                width: '200px',
                height: '200px',
                objectFit: 'scale-down'
              }}
              src={imageUrl}
              ui={false}
            />
            <Card.Header textAlign='center'>
              <Link
                onClick={() => fetchCurrentPodcastConnect(rss)}
                to='/podcast'
              >
                {name}
              </Link>
            </Card.Header>
          </Card.Content>
        </Card>
      ))}
    </Card.Group>
  ) : (
    <Dimmer active inverted>
      <Loader size='massive'>Loading</Loader>
    </Dimmer>
  );
}

const dispatchToProps = dispatch => {
  return {
    fetchCurrentPodcastConnect: rss => dispatch(fetchCurrentPodcast(rss))
  };
};

const ConnectedPodcastList = connect(null, dispatchToProps)(PodcastList);

export default ConnectedPodcastList;

PodcastList.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  podcasts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      image: PropTypes.string,
      rss: PropTypes.string,
      name: PropTypes.string
    })
  ).isRequired,
  fetchCurrentPodcastConnect: PropTypes.func.isRequired
};
