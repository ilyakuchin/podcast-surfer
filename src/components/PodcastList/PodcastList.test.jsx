import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PodcastList } from './PodcastList';

configure({ adapter: new Adapter() });

describe('Test PodcastList component', () => {
  test('should render correctly', () => {
    expect(
      toJson(
        shallow(
          <PodcastList
            isFetching={false}
            jwt='jwt'
            fetchCurrentPodcastConnect={() => {}}
            podcasts={[
              { id: '1', image: 'link', rss: 'rss1', name: 'name' },
              { id: '2', image: 'link2', rss: 'rss2', name: 'name2' },
              { id: '3', image: 'link3', rss: 'rss3', name: 'name3' }
            ]}
          />
        )
      )
    ).toMatchSnapshot();
  });

  test('should render correctly if Fecthing', () => {
    expect(
      toJson(
        shallow(
          <PodcastList
            isFetching
            jwt='jwt'
            fetchCurrentPodcastConnect={() => {}}
            podcasts={[
              { id: '1', image: 'link', rss: 'rss1', name: 'name' },
              { id: '2', image: 'link2', rss: 'rss2', name: 'name2' },
              { id: '3', image: 'link3', rss: 'rss3', name: 'name3' }
            ]}
          />
        )
      )
    ).toMatchSnapshot();
  });
});
