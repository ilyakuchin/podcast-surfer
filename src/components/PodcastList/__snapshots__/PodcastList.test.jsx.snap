// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PodcastList component should render correctly 1`] = `
<CardGroup>
  <Card
    centered={true}
    key="rss1"
  >
    <CardContent
      textAlign="center"
    >
      <Image
        as="img"
        style={
          Object {
            "height": "200px",
            "objectFit": "scale-down",
            "width": "200px",
          }
        }
        ui={false}
      />
      <CardHeader
        textAlign="center"
      >
        <Link
          onClick={[Function]}
          to="/podcast"
        >
          name
        </Link>
      </CardHeader>
    </CardContent>
  </Card>
  <Card
    centered={true}
    key="rss2"
  >
    <CardContent
      textAlign="center"
    >
      <Image
        as="img"
        style={
          Object {
            "height": "200px",
            "objectFit": "scale-down",
            "width": "200px",
          }
        }
        ui={false}
      />
      <CardHeader
        textAlign="center"
      >
        <Link
          onClick={[Function]}
          to="/podcast"
        >
          name2
        </Link>
      </CardHeader>
    </CardContent>
  </Card>
  <Card
    centered={true}
    key="rss3"
  >
    <CardContent
      textAlign="center"
    >
      <Image
        as="img"
        style={
          Object {
            "height": "200px",
            "objectFit": "scale-down",
            "width": "200px",
          }
        }
        ui={false}
      />
      <CardHeader
        textAlign="center"
      >
        <Link
          onClick={[Function]}
          to="/podcast"
        >
          name3
        </Link>
      </CardHeader>
    </CardContent>
  </Card>
</CardGroup>
`;

exports[`Test PodcastList component should render correctly if Fecthing 1`] = `
<Dimmer
  active={true}
  inverted={true}
>
  <Loader
    size="massive"
  >
    Loading
  </Loader>
</Dimmer>
`;
